### ISSUE TITLE
CP-SAT Occasional Segmentation Fault

### ISSUE THREAD

**Issue by willf-j** [google/or-tools#4400]
**What version of OR-Tools and what language are you using?**
Version: 9.11.4210
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT

**What operating system (Linux, Windows, ...) and version?**
Linux; Ubuntu 22.04

**What did you do?**
Run the basic python script to load and run the model

**What did you expect to see**
Model runs without issue; will converge after a few minutes to an objective around 30000. Certificate of optimality takes rather a lot longer on this model

**What did you see instead?**
Segmentation fault within the first ~20 seconds after presolve. Happens about 10% of the time.

Make sure you include information that can help us debug (full error message, model Proto).
```
user$ python run_model.py cp_sat_model.proto
Model loaded successfully from cp_sat_model.proto

Starting CP-SAT solver v9.11.4210
Parameters: log_search_progress: true
Setting number of workers to 32
.
.
.
#Bound   9.43s best:inf   next:[19048,10000000] fs_random
#Bound   9.54s best:inf   next:[19088,10000000] fs_random
#Bound   9.66s best:inf   next:[19116,10000000] fs_random
#Bound  11.17s best:inf   next:[19130,10000000] fs_random
#Model  11.99s var:2265/2267 constraints:17457/17458
#Model  13.12s var:2263/2267 constraints:17456/17458
#Bound  13.74s best:inf   next:[20369,10000000] objective_lb_search
#1      14.15s best:55582 next:[20369,55581] rens_lp_lns (d=0.97 s=772 t=0.10 p=1.00 stall=0 h=auto_l0)
Segmentation fault (core dumped)

```
[crashing_model.zip](https://github.com/user-attachments/files/17265139/crashing_model.zip)

**Anything else we should know about your project / environment**
13th Gen Intel® Core™ i9-13900K × 32

**Comment by willf-j** [comment]
I think it may be easier to repro if you set `solver.parameters.use_lns_only = True`

**Comment by lperron** [comment]
I ran it for 3 days on main. No crash. I will close it.

**Comment by lperron** [comment]
I ran it manually around 100 times. It never crashed.
And I know we fixed a few crashes post 9.11.

I will continue trying.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00



Le mar. 8 oct. 2024 à 18:04, Will Floyd-Jones ***@***.***> a
écrit :
